### PR TITLE
Add cryptosuite instantiation section.

### PR DESCRIPTION
**Pull request by msporny**
This PR aligns the specification with the new cryptosuite interface introduced by PR https://github.com/w3c/vc-data-integrity/pull/244.

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/vc-di-eddsa/pull/77.html" title="Last updated on Apr 28, 2024, 2:51 PM UTC (4e7ac43)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/vc-di-eddsa/77/51b7d90...4e7ac43.html" title="Last updated on Apr 28, 2024, 2:51 PM UTC (4e7ac43)">Diff</a>

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -441,45 +493,118 @@ <h3>eddsa-rdfc-2022</h3>
...
+          <li>
+Let |verified:boolean| be the result of running the algorithm in Section
+[[[#proof-verification-eddsa-rdfc-2022]]] algorithm on |hashData|,
+|proofBytes|, and |proofConfig|.
+            </li>
+            <li>
+If |proof|.|created| does not [=map/exist=],
+an error MUST be raised and SHOULD convey an error type of
+<a data-cite="VC-DATA-INTEGRITY#MALFORMED_PROOF_ERROR">
+MALFORMED_PROOF_ERROR</a>.
```
Comment (dlongley) [inline index.html]: `created` is optional, this should only be checking that the format of the value is proper if `created` exists.

**Review by dlongley — CHANGES_REQUESTED** [review]

File: index.html (hunk 1 above)
Comment (dlongley) [reply index.html]: Also, that check is already done in the proof configuration section:

> Set proofConfig.created to options.created. If the value is not a valid [[XMLSCHEMA11-2](https://w3c.github.io/vc-di-eddsa/#bib-xmlschema11-2)] datetime, an INVALID_PROOF_DATETIME error MUST be raised.

https://w3c.github.io/vc-di-eddsa/#proof-configuration-eddsa-rdfc-2022

So maybe it doesn't need repeating here.

**Review by Wind4Greg — APPROVED** [review]
Concur with @dlongley on handling of the `created` field.

File: index.html (hunk 1 above)
Comment (msporny) [reply index.html]: Fixed in 643bffc1687d57e3901d24a37e8f5c405fbf889f.

File: index.html [hunk 2]
```diff
@@ -441,45 +493,112 @@ <h3>eddsa-rdfc-2022</h3>
...
+          </li>
+          <li>
+Let |transformedData| be the result of running the algorithm in Section <a
+href="#transformation-eddsa-rdfc-2022"></a> with |unsecuredDocument| and
+|proofConfig| passed as parameters.
+          </li>
+          <li>
+Let |hashData| be the result of running the algorithm in Section
+[[[#hashing-eddsa-rdfc-2022]]] with |transformedData| and |proofConfig|
+passed as a parameters.
```
Comment (dlongley) [inline index.html:586]: Hmm, the hashing algorithm takes the canonical proof config, not what is really "proof options" here under the name `proofConfig`. I'll make some suggestions above to try and remedy this.

File: index.html [hunk 3]
```diff
@@ -441,45 +493,112 @@ <h3>eddsa-rdfc-2022</h3>
...
+          </dl>
+
+          <ol class="algorithm">
+          <li>
+Let |unsecuredDocument| be a copy of |securedDocument| with
+the `proof` value removed.
+          </li>
+          <li>
+Let |proofConfig| be a copy of |securedDocument|.|proof| with `proofValue`
+removed.
```
Comment (dlongley) [inline index.html]:
```suggestion
Let |proofOptions| be the result of a copy of |securedDocument|.|proof| with `proofValue`
removed.
```

File: index.html [hunk 4]
```diff
@@ -441,45 +493,112 @@ <h3>eddsa-rdfc-2022</h3>
...
+          </li>
+          <li>
+Let |proofBytes| be the
+<a data-cite="VC-DATA-INTEGRITY#multibase-0">Multibase decoded base58-btc
+value</a> in |securedDocument|.|proof|.|proofValue|.
+          </li>
+          <li>
+Let |transformedData| be the result of running the algorithm in Section <a
+href="#transformation-eddsa-rdfc-2022"></a> with |unsecuredDocument| and
+|proofConfig| passed as parameters.
```
Comment (dlongley) [inline index.html]:
```suggestion
|proofOptions| passed as parameters.
```

File: index.html [hunk 5]
```diff
@@ -441,45 +493,112 @@ <h3>eddsa-rdfc-2022</h3>
...
+          </li>
+          <li>
```
Comment (dlongley) [inline index.html]:
```suggestion
          </li>
Let |proofConfig| be the result of running the algorithm in Section <a
href="#proof-configuration-eddsa-rdfc-2022"></a> with |unsecuredDocument| and
|proofOptions| passed as parameters.
          <li>
```

**Review by dlongley — CHANGES_REQUESTED** [review]
I think we need a couple more changes here (please check my work, cc: @Wind4Greg as well) to get the algorithm inputs/outputs to line up properly. If my work is correct, then I don't know if we have a similar issue with the recent changes in the ecdsa cryptosuite.

EDIT: We do -- we need an editorial fix over in the ecdsa cryptosuites spec for this as well, it's the same issue there.

File: index.html (hunk 2 above)
Comment (msporny) [reply index.html:586]: Suggestions have been applied.

**Comment by msporny** [comment]
Normative, multiple reviews, changes requested and made, no objections, merging.

**Pull request merged**